### PR TITLE
Fix longjmp on Windows

### PR DESCRIPTION
**Pull request by patrickbkr**
See https://blog.lazym.io/2020/09/21/Unicorn-Devblog-setjmp-longjmp-on-Windows/ for a good explanation on what we're doing here.

### CODE REVIEW

**Comment by raiph** [comment]
Iirc the JIT has been disabled by default for main head with no plan to re-enable it by default. Is my memory about right and so this fix is "just" for those who decide to nevertheless manually re-enable it or am I misremembering or misunderstanding something?

**Comment by patrickbkr** [comment]
Moar has long had two JITs. The LegoJIT and the ExpressionJIT. We disabled - and plan to remove - the ExprJIT. The LegoJIT stays.

File: src/platform/setjmp.h [hunk 1]
```diff
@@ -1,6 +1,28 @@
-/* MinGW x64 BUGFIX: Pass a NULL argument instead of the frame pointer to the
- * setjmp implementation to make the JIT not choke on exceptions on MinGW.
- * cygx++ for fix */
+/* In later Windows versions the longjmp implementation tries to perform stack
+ * unwinding (for the purposes of cleanup). This is problematic for MoarVM
+ * because our JIT produces frames that this stack unwinder doesn't understand.
+ * Luckily the longjmp function can be instructed to not do the stack unwinding
+ * by setting a respective field in the jmpbuf structure. The setjmp function
+ * does so when it's *second* argument is NULL. (The frame pointer) Microsoft
```
Comment (MasterDuke17) [inline src/platform/setjmp.h:6]: `(The frame pointer)` seems kind of out of place?

File: src/platform/setjmp.h [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+ * the function is marked as purely intrinsic and the compiler won't let you
+ * take the address of such a function. So we instead modify the resulting
+ * jmpbuf and set that flag ourself. Microsofts `setjmp.h` header file has
+ * proven helpful in disecting the otherwise opaque jmp_buf struct.
+ * All of the above (both, the two arg setjmp we use on MinGW and modifying
+ * the jmpbuf we do on MSVC) is not part of any public Windows API and could in
+ * principle break any time. Given that
+ * - Microsoft has a long history of being very careful to keep back-compat and
+ * - these hacks are in use by MinGW and by extension some high profile
+ *   projects (e.g. QEmu) and
```
Comment (MasterDuke17) [inline src/platform/setjmp.h:20]: I believe 'QEMU' is correct (i.e., all caps).

**Pull request merged**